src/Streaming/Internal.hs

```diff
@@ -1098,7 +1172,10 @@ four<Enter>
 
 -}
 never :: (Monad m, Applicative f) => Stream f m r
-never =  let loop = Effect $ return $ Step $ pure loop in loop
+-- The Monad m constraint is only used for compatibility with FreeT.
+-- Even then, it should really be an Applicative constraint, but
+-- this library still supports older versions of base.
+never =  let loop = Step $ pure loop in loop
 {-#INLINABLE never #-}
 
```

Comment:
The modified `never` is wrong! If `pure` is strict, as it is for `Identity`, then everything blows up. Argh. I guess I'll pull this change out. It's a bit sad, and makes me wonder if there's some horribly evil way to avoid it.

Reply:
In particular, suppose we had a copy of the `Stream` type with a lazy `StepL` constructor, and we wrote

```haskell
never = unsafeCoerce loop where
  loop = StepL $ pure loop
```

Would this go wrong?